core/src/impl/Kokkos_Stacktrace.cpp

```diff
@@ -0,0 +1,265 @@
+  for (auto&& entry : traceback) {
+    size_t col_count = 0;
+    for_each_token (entry, [&] (const std::string& s, bool) {
+                             const size_t cur_col_len =
+                               (found_main && col_count == main_col) ?
+                               demangle (s).size () :
+                               s.size ();
+                             ++col_count;
+                             if (max_col_lens.size () < col_count) {
+                               max_col_lens.push_back (cur_col_len);
+                             }
+                             else {
+                               const size_t old_max_len = max_col_lens[col_count-1];
+                               if (old_max_len < cur_col_len) {
+                                 max_col_lens[col_count-1] = cur_col_len;
+                               }
+                             }
+                           });
+  }
+  return std::make_tuple(found_main, main_col, max_col_lens);
+}
+
+void
+demangle_and_print_traceback_entry (std::ostream& out,
+                                    const std::string& traceback_entry,
+                                    const bool found_main,
+                                    const size_t main_col,
+                                    const std::vector<size_t>& max_col_lens)
+{
+  std::vector<std::string> tokens;
+  size_t cur_col = 0;
+  for_each_token (traceback_entry,
+                  [&] (const std::string& s, bool last) {
+                    const size_t old_width (out.width ());
+                    out.width (max_col_lens[cur_col]);
+                    try {
+                      if (found_main && cur_col == main_col) {
+                        out << demangle (s);
+                      }
+                      else {
+                        out << s;
+                      }
+                      if (! last) {
+                        out << " ";
+                      }
+                      ++cur_col;
+                    }
+                    catch (...) {
+                      out.width (old_width);
+                      throw;
+                    }
+                    out.width (old_width);
+                  });
+}
+
+void
+demangle_and_print_traceback (std::ostream& out,
+                              const std::vector<std::string>& traceback)
+{
+  const auto result = find_main_column (traceback);
+  for (auto&& entry : traceback) {
+    using std::get;
+    demangle_and_print_traceback_entry (out, entry, get<0> (result),
+                                        get<1> (result), get<2> (result));
+    out << std::endl;
+  }
+}
+
+void
+print_saved_stacktrace (std::ostream& out)
+{
+  auto lines = Stacktrace::lines ();
+  for (auto&& entry : lines) {
+    out << entry << std::endl;
+  }
+}
+
+void
+print_demangled_saved_stacktrace (std::ostream& out)
+{
+  demangle_and_print_traceback (out, Stacktrace::lines ());
+}
+
+std::function<void()> user_terminate_handler_post_;
```

Comment:
:(

Reply:
Global variable.

Reply:
In https://github.com/kokkos/kokkos/pull/2226/commits/2bed7f2c4b14e148b976578a465b9d9c6261aef1 it is still a global variable, but it is now initialized to nullptr